src/main/java/com/direwolf20/buildinggadgets/common/items/gadgets/GadgetDestruction.java

```diff
@@ -259,219 +206,152 @@ public static void anchorBlocks(EntityPlayer player, ItemStack stack) {
+
         if (GadgetGeneric.getConnectedArea(stack)) {
-            String[] directionNames = NBTKeys.GADGET_VALUES;
-            AxisAlignedBB area = new AxisAlignedBB(pos);
-            for (int i = 0; i < directionNames.length; i++)
-                area = area.union(new AxisAlignedBB(pos.offset(directions.get(i), getToolValue(stack, directionNames[i]) - (i == 4 ? 1 : 0))));
+            HashSet<BlockPos> voidPositions = new HashSet<>();
+            int depth = getToolValue(stack, NBTKeys.GADGET_VALUE_DEPTH);
+            if (depth == 0)
+                return new SetBackedPlacementSequence(voidPositions, boundary);
 
-            addConnectedCoords(world, player, startPos, stateTarget, voidPositions,
-                    (int) area.minX, (int) area.minY, (int) area.minZ, (int) area.maxX - 1, (int) area.maxY - 1, (int) area.maxZ - 1);
-        } else {
-            int left = -getToolValue(stack, NBTKeys.GADGET_VALUE_LEFT);
-            int right = getToolValue(stack, NBTKeys.GADGET_VALUE_RIGHT);
-            int down = -getToolValue(stack, NBTKeys.GADGET_VALUE_DOWN);
-            int up = getToolValue(stack, NBTKeys.GADGET_VALUE_UP);
-            for (int d = 0; d < depth; d++) {
-                for (int x = left; x <= right; x++) {
-                    for (int y = down; y <= up; y++) {
-                        BlockPos voidPos = new BlockPos(startPos);
-                        voidPos = voidPos.offset(directions.get(0), x);
-                        voidPos = voidPos.offset(directions.get(2), y);
-                        voidPos = voidPos.offset(directions.get(4), d);
-                        if (validBlock(world, voidPos, player, stateTarget))
-                            voidPositions.add(voidPos);
-                    }
-                }
-            }
+            addConnectedCoordinates(world, player, startPos, stateTarget, voidPositions, boundary);
+            return new SetBackedPlacementSequence(voidPositions, boundary);
         }
-        return voidPositions;
+
+        return new SetBackedPlacementSequence(boundary.stream()
+                .filter(p -> isValidBlock(world, p, player, stateTarget))
+                .collect(Collectors.toCollection(HashSet::new)), boundary);
     }
 
-    public static void addConnectedCoords(World world, EntityPlayer player, BlockPos loc, IBlockState state,
-            SortedSet<BlockPos> coords, int minX, int minY, int minZ, int maxX, int maxY, int maxZ) {
-        if (coords.contains(loc) || loc.getX() < minX || loc.getY() < minY || loc.getZ() < minZ || loc.getX() > maxX || loc.getY() > maxY || loc.getZ() > maxZ)
-            return;
+    public static List<BlockPos> getClearingPositionsForRendering(World world, BlockPos pos, EnumFacing incomingSide, EntityPlayer player, ItemStack stack) {
+        ArrayList<BlockPos> list = getClearingPositions(world, pos, incomingSide, player, stack).stream()
```

Comment:
```suggestion
        List<BlockPos> list = getClearingPositions(world, pos, incomingSide, player, stack).stream()
```
I know it's a local variable, but please avoid exact Types throught Code. You should only be using the interfaces.

Reply:
There are also other occurences, I won't add other review comments for them, but please fix them too.

Reply:
What lines are the other occurrences on? I can't seem to find them.